lec17-rmarkdown.Rmd

```diff
@@ -1,4 +1,306 @@
+```
+- Unordered list item
+    - Unordered list sub-item
+- Unordered list item
+```
+- Unordered list item
+    - Unordered list sub-item
+- Unordered list item
+
+**Ordered Lists (i.e. numbers)**
+```
+1. Ordered list item
+    1. Ordered list sub-item
+    2. Ordered list sub-item
+2. Ordered list item
+```
+1. Ordered list item
+    1. Ordered list sub-item
+    2. Ordered list sub-item
+2. Ordered list item
+
+#### Tables
+The `knitr` package has a function called `kable` that helps to display tables
+from an `r` code chunk nicely. It is best to use `echo=FALSE` and `results='asis'`.
+
+`r ''````{r, echo=FALSE, results='asis'}
+library(knitr)
+kable(head(mtcars))
+```
+
+You can also set the default data frame printing via the `df_print` option in
+your YAML metadata under `output` to do this automatically.
+
+```
+---
+title: Document
+output:
+    html_document:
+        df_print:kable
+---
+```
+
+#### In-line code
+
+If you want to state a value in your data in your text, it is best to reference
+to the actual variable or code containing that value rather than manually writing
+it out. Here is an example:
+
+There are `r ''``r nrow(df)` samples in this experiment.
+
+
+### Images and Figures
+To include an image, use the following syntax:
+```
+[caption for my image](path/to/image.png)
+```
+If you would like the caption you wrote to be included underneath your image,
+put the following in your YAML metadata:
+
+```
+---
+title: Document
+output:
+    html_document:
+        fig_caption: yes
+
+```
+As you've learned in doing your assignments, both the code and output of your
+`r` code chunk such as figures will show up in your output document. However,
+using code chunk options `echo` and `eval`, you can suppress output of the code
+underlying a graph, and only show the resulting plot from your code or vice versa.
+In this case, you probably want the former, and it would look like this:
+
+`r ''````{r, eval=TRUE, echo=FALSE}
+library(ggplot2)
+qplot(mpg, wt, data=mtcars)
+```
+
+You can also set the figure height and width too using the `fig.height` and
+`fig.width` options.
+
+`r ''````{r, fig.width=7, fig.height=7, eval=TRUE, echo=FALSE}
+library(ggplot2)
+qplot(mpg, wt, data=mtcars)
+```
+
+### In-Line Citations & Bibliography
+
+Pandoc can automatically generate citations and a bibliography in a number of
+styles. In order to use this feature, you will need to specify a bibliography
+file using the bibliography metadata field in a YAML metadata section.
+For example:
+
+```
+---
+title: "Sample Document"
+output: html_document
+bibliography: bibliography.bib
+---
+```
+
+Many bibliography formats are accepted (see the [R Markdown guides](http://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html)),
+such as a .bib file which many citation managers can generate for you and will
+hold all the references you need for your document (e.g. [EndNote](https://www.reed.edu/cis/help/LaTeX/EndNote.html))
```

Comment:
I think we should give Zotero (or other open source reference managers such as JabRef and Doccear (and could also have been Mendeley before they sold their soul to Elsevier)) as the primary examples. Commercial ones, like EndNote, can of course be mentioned, but I have yet to see anything they have to offer that would make me switch over (especially not in the context of exporting to bibtex, and certainly not to EndNote (which I have admittedly never used, but looks like Windows 1995 and runs (crawls) like it was designed by Adobe, from what I have seen from others)).